streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -314,6 +319,42 @@ private void shutdownTasksAndState(final boolean rethrowExceptions) {
         }
     }
 
+    /**
+     * Similar to shutdownTasksAndState, however does not close the task managers,
+     * in the hope that soon the tasks will be assigned again
+     * @param rethrowExceptions
+     */
+    private void suspendTasksAndState(final boolean rethrowExceptions) {
+        log.debug("{} suspendTasksAndState: suspending all active tasks [{}] and standby tasks [{}]", logPrefix,
+            activeTasks.keySet(), standbyTasks.keySet());
+
+        // Commit first as there may be cached records that have not been flushed yet.
+        commitOffsets(rethrowExceptions);
+        // flush state
+        flushAllState(rethrowExceptions);
+        // flush out any extra data sent during close
+        producer.flush();
```

Comment:
There is no harm in doing it here, but it also needs to happen after the tasks are closed.